src/completion.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const stdx = @import("stdx.zig");
+const assert = std.debug.assert;
+
+const RingBuffer = @import("ring_buffer.zig").RingBufferType;
+const BoundedArray = stdx.BoundedArrayType;
+
+const completion_entries = 256;
+const completion_entry_bytes = 512;
+const keywords = [_][]const u8{
+    "create_accounts",
+    "create_transfers",
+    "lookup_accounts",
+    "lookup_transfers",
+    "get_account_transfers",
+    "get_account_balances",
+    "query_accounts",
+    "query_transfers",
+    "help",
+    "id",
+    "code",
+    "ledger",
+    "flags",
+    "account_id",
+    "debit_account_id",
+    "credit_account_id",
+    "amount",
+};
+
+pub const Completion = struct {
+    const CompletionList = RingBuffer([completion_entries]u8, .slice);
```

Comment:
I _think_ this should be `RingBuffer([completion_entry_bytes]u8 ...`, not `[completion_entries]u8` - the type of element in the ringbuffer is a completion entry.

Likewise, I _think_ the wrong constant is used for `CompletionList.init` in `Completion.init` - there we should be asking for the number of elements - `completion_entries`.

Reply:
fixed